core/lpms_test.go

```diff
@@ -0,0 +1,109 @@
+package core
+
+import (
+	"testing"
+	"os"
+	"fmt"
+	"github.com/livepeer/lpms/stream"
+	"time"
+	"github.com/livepeer/lpms/segmenter"
+	"context"
+	"github.com/nareix/joy4/av"
+	"github.com/nareix/joy4/format"
+	"github.com/nareix/joy4/av/avutil"
+	"github.com/golang/glog"
+	"io"
+	"github.com/ericxtang/m3u8"
+	"github.com/livepeer/lpms/vidplayer"
+	"net/url"
+	"github.com/nareix/joy4/format/rtmp"
+)
+
+type TestStream struct{}
+
+func (s TestStream) String() string                       { return "" }
+func (s *TestStream) GetStreamFormat() stream.VideoFormat { return stream.RTMP }
+func (s *TestStream) GetStreamID() string                 { return "test" }
+func (s *TestStream) Len() int64                          { return 0 }
+func (s *TestStream) ReadRTMPFromStream(ctx context.Context, dst av.MuxCloser) (chan struct{}, error) {
+	format.RegisterAll()
+	wd, _ := os.Getwd()
+	file, err := avutil.Open(wd + "/test.flv")
+	if err != nil {
+		fmt.Println("Error opening file: ", err)
+		return nil, err
+	}
+	header, err := file.Streams()
+	if err != nil {
+		glog.Errorf("Error reading headers: %v", err)
+		return nil, err
+	}
+
+	dst.WriteHeader(header)
+	eof := make(chan struct{})
+	go func(eof chan struct{}) {
+		for {
+			pkt, err := file.ReadPacket()
+			if err == io.EOF {
+				dst.WriteTrailer()
+				eof <- struct{}{}
+			}
+			dst.WritePacket(pkt)
+		}
+	}(eof)
+	return eof, nil
+}
+func (s *TestStream) WriteRTMPToStream(ctx context.Context, src av.DemuxCloser) (chan struct{}, error) {
+	return nil, nil
+}
+func (s *TestStream) WriteHLSPlaylistToStream(pl m3u8.MediaPlaylist) error                { return nil }
+func (s *TestStream) WriteHLSSegmentToStream(seg stream.HLSSegment) error                 { return nil }
+func (s *TestStream) ReadHLSFromStream(ctx context.Context, buffer stream.HLSMuxer) error { return nil }
+func (s *TestStream) ReadHLSSegment() (stream.HLSSegment, error)                          { return stream.HLSSegment{}, nil }
+func (s *TestStream) Width() int                                                          { return 0 }
+func (s *TestStream) Height() int                                                         { return 0 }
+func (s *TestStream) Close()															  {}
+
+func TestRetryRTMPToHLS(t *testing.T) {
```

Comment:
Thanks for the work on testing! We can probably simplify this substantially by stubbing out the segmenter. That way, we can avoid instantiating so much networking and file IO.

This also allows us to ensure that the retries are actually being called, by counting the invocations to the stubbed `segmenter.RTMPToHLS`.  Note that the `segmenter.VideoSegmenter` interface (currently empty) will need the RTMPToHLS function added to it.

Does that make sense?

Reply:
Do u mean creating a test case for rtmpToHLS static method, with a TestVideoSegmenter. Fail it first two times and verify that it is being attempted thrice?

I have made changes as per that, but I am not sure if this is what you meant. Please confirm.